PWGJE/Tasks/nucleiWithFastJet.cxx

```diff
@@ -0,0 +1,1500 @@
+        if (jetFlag) {
+          jetHist.fill(HIST("mcpJet/triton/pt/PtTriton_jet"), mcParticle.pt());
+        } else {
+          jetHist.fill(HIST("mcpJet/triton/pt/PtTriton_outJet"), mcParticle.pt());
+        }
+      }
+
+      if (mcParticle.pdgCode() == -PDGTriton) {
+        jetHist.fill(HIST("mcpJet/antiTriton/pt/PtantiTriton"), mcParticle.pt());
+        if (jetFlag) {
+          jetHist.fill(HIST("mcpJet/antiTriton/pt/PtantiTriton_jet"), mcParticle.pt());
+        } else {
+          jetHist.fill(HIST("mcpJet/antiTriton/pt/PtantiTriton_outJet"), mcParticle.pt());
+        }
+      }
+    } // track
+  }   // process mc
+
+  void processMCRec(o2::aod::JCollision const& collisionJet, soa::Join<aod::JTracks, aod::JTrackPIs, aod::McTrackLabels> const& tracks,
+                    soa::Filtered<aod::ChargedMCDetectorLevelJets> const& mcdjets, aod::McParticles const&, TrackCandidatesMC const&)
+  {
+    jetHist.fill(HIST("mcdJet/eventStat"), 0.5);
+    // JEhistos.fill(HIST("nEvents_MCRec"), 0.5);
+
+    if (!jetderiveddatautilities::selectCollision(collisionJet, jetderiveddatautilities::JCollisionSel::sel8))
+      return;
+    // bool jetFlag = kFALSE;
+
+    jetHist.fill(HIST("mcdJet/eventStat"), 1.5);
+
+    if (fabs(collisionJet.posZ()) > 10)
+      return;
+
+    jetHist.fill(HIST("mcdJet/eventStat"), 2.5);
+    bool INELgt0 = false;
+    for (const auto& track : tracks) {
+      if (fabs(track.eta()) < cfgtrkMaxEta) {
+        INELgt0 = true;
+        break;
+      }
+    }
+    if (!INELgt0)
+      return;
+    jetHist.fill(HIST("mcdJet/eventStat"), 3.5);
+    int nJets = 0;
+    for (auto& mcdjet : mcdjets) {
+      jetHist.fill(HIST("mcdJet/hJetPt"), mcdjet.pt());
+      jetHist.fill(HIST("mcdJet/hJetEta"), mcdjet.eta());
+      jetHist.fill(HIST("mcdJet/hJetPhi"), mcdjet.phi());
+      nJets++;
+    }
+
+    jetHist.fill(HIST("mcdJet/vertexZ"), collisionJet.posZ());
+    jetHist.fill(HIST("mcdJet/nJetsPerEvent"), nJets);
+
+    for (const auto& track : tracks) {
+      auto fullTrack = track.track_as<TrackCandidatesMC>();
+
+      if (!isTrackSelected(fullTrack))
+        continue;
+      if (fabs(fullTrack.eta()) > cfgtrkMaxEta)
+        continue;
+      if (!track.has_mcParticle())
+        continue;
+      if (!track.mcParticle().isPhysicalPrimary())
+        continue;
+      if (fabs(track.mcParticle().y()) > cfgtrkMaxRap)
+        continue;
+
+      bool jetFlag = false;
+      // float jetPt = -999.;
+
+      for (auto& mcdjet : mcdjets) {
+        double delPhi = TVector2::Phi_mpi_pi(mcdjet.phi() - track.phi());
+        double delEta = mcdjet.eta() - track.eta();
+        double R = TMath::Sqrt((delEta * delEta) + (delPhi * delPhi));
+        if (R < cfgjetR)
+          jetFlag = true;
+        //  jetPt = mcdjet.pt();
+        break;
+      } // jet
+
+      if (track.mcParticle().pdgCode() == PDGProton) {
+        jetHist.fill(HIST("mcdJet/proton/pt/PtProton"), track.mcParticle().pt());
```

Comment:
you can probably replace all these histograms below and all the if conditions with just one 3D histogram with axes of pt, jetFlag and pdgCode where you use an enum to map the pdg codes yourself. Then you have 1 histogram and just one line like this

          jetHist.fill(HIST("mcdJet/pt/PtPatritcleType"), track.mcParticle().pt(), jetFlag, mapPDGCodes(track.mcParticle().pdgCode()));
          
          This also applies to the other process functions above as well

Reply:
Thanks for the suggestion. I have now implemented a mapping for the hist in the MC process functions.